### ISSUE TITLE
Suggestion: Passing observable store as props

### ISSUE THREAD

**Issue by ro-savage** [solkimicreb/react-easy-state#23]
You may have already thought of this / decided it was a bad idea. My understanding of how all this works it pretty damn basic (never used observables or MobX/Reactive Framework)

What if you passed the observable as a prop to the component, so that you can still use propTypes and this behave a bit more as 'normal' react.

For example
```js
import React from 'react'
import PropTypes from 'prop-types'
import { view } from 'react-easy-state'
import destinations from '../destinationsStore.js'

const ListDestination = (props) => {
  return (
    <ul>
      {props.destinations.list.map(destination => {
        return (
          <li key={destination}>{destination}</li>
        )
      })}
    </ul>
  )
}

ListDestination.propTypes = {
  destinations: PropTypes.shape({
    list: PropTypes.array,
  }),
}

export default view(ListDestination, {destinations})
```

**Comment by ro-savage** [comment]
A naive implementation for stateless components, from hacking away at my node_modules `es.es5.js` file

```js
function view(Comp, obs) {
  var isStatelessComp = !(Comp.prototype && Comp.prototype.isReactComponent);
  /* ... */
  ReactiveHOC.prototype.render = function render () {
      const obsProps = Object.assign({}, this.props, obs)
      return isStatelessComp ? Comp(obsProps, this.context) : BaseComp.prototype.render.call(this);
    };
}
```

**Comment by solkimicreb** [comment]
This kind of mixes props with state. I am not sure about the benefits, but I try to understand.

What do you mean by this?

> this behaves a bit more as 'normal' react.

**Comment by ro-savage** [comment]
In react, a component re-renders based on `State` (internal to a component) or `Props` ( external to a component).

The `react-easy-state` store is **external** to the component because it can be used in multiple components. Therefore I would usually expect the data to come through props.

This is how Redux works and I believe MobX, freactal, xstate and others.

It also keeps with the one-way data flow of React. Now I can, for example, log all my props and see the changes that caused the re-render.

**Comment by ro-savage** [comment]
Having said that. 
To quote React's Docs: **Props are Read-Only**
Props are usually considered read-only and in this case they don't have to be read-only. 

However, React components are usually considered **pure functions** and therefore accessing and manipulating data that is not passed directly to the component is also not 'normal' react.

Pass them as props *feels* to me more like I would expect React to work, and would allow easier transition between `react-easy-state` and `Redux`/`MobX`.  You could get away with minimal changes to individual components.

**Comment by solkimicreb** [comment]
Using props the access state is just one flavor of MobX and Redux (with the `<Provider />`) pattern. I think the primary reason why there are so numerous state management libs for React is the fact that it is missing external state management (and a nice way of sharing state between sibling components).

I think that external state is 'none React', but it something really needed. The provider pattern is pretty nice though when you have none singleton (local state) that you want to pass down deep into your component tree. Easy State has no solution - like MobX and Redux - for these cases. I think that stores like this can be moved to the global space 99.9% of the time and they can be used as an external state (with simple `import store from './myStore'` instead of a complex context based `Provider` pattern).

In a nutshell: I would wait with changes like this until we see if people has DI problems or not.
